code/modules/clothing/glasses/glasses.dm

```diff
@@ -495,4 +495,48 @@
 		else
 			to_chat(user, "<span class='notice'>The eye winks at you and vanishes into the abyss, you feel really unlucky.</span>")
 		qdel(src)
-	..()
\ No newline at end of file
+	..()
+
+/obj/item/clothing/glasses/tajblind
+	name = "embroidered veil"
+	desc = "An Ahdominian made veil that allows the user to see while obscuring their eyes."
+	icon_state = "tajblind"
+	item_state = "tajblind"
+	flags_inv = HIDEEYES
+	actions_types = list(/datum/action/item_action/toggle)
+	up = 0
+	tint = 0
+
+/obj/item/clothing/glasses/tajblind/eng
+	name = "industrial veil"
+	icon_state = "tajblind_engi"
+	item_state = "tajblind_engi"
+
+/obj/item/clothing/glasses/tajblind/sci
+	name = "hi-tech veil"
+	icon_state = "tajblind_sci"
+	item_state = "tajblind_sci"
+
+/obj/item/clothing/glasses/tajblind/cargo
+	name = "khaki veil"
+	icon_state = "tajblind_cargo"
+	item_state = "tajblind_cargo"
+
+/obj/item/clothing/glasses/tajblind/attack_self()
+	toggle_veil()
+
+/obj/item/clothing/glasses/proc/toggle_veil()
+	if(usr.canmove && !usr.incapacitated())
+		if(up)
+			up = !up
+			icon_state = initial(icon_state)

```

Comment:
I mentioned this to you last night, but am posting it again just in case you forgot.

Both the toggle states call `icon_state = initial(icon_state)` which means the icon_state always appears the same regardless of activation. Might want to adjust that unless it is intentional, in which case you can just remove the icon_state changes as they are unnecessary.